test/nbb/main_test.cljs

```diff
@@ -193,6 +194,27 @@
       (.then (fn [val]
                (is (= 2 val))))))
 
+(deftest-async gobject-get-all-property-names
+  (-> (nbb/load-string "(ns foo (:require [goog.object :as gobj]))
+                        (gobj/getAllPropertyNames #js{:x 0 :y 1 :z 2})")
+      (.then (fn [val]
+               (is (= ["x" "y" "z"] (js->clj val)))))))
+
+(deftest-async gobject-remove
+  (-> (nbb/load-string "(ns foo (:require [goog.object :as gobj]))
+                        (def obj #js {:x 0 :y 1})
+                        (gobj/remove obj \"x\")
+                        obj")
+      (.then (fn [val]
+               (is (= {"y" 1} (js->clj val)))))))
+
+(deftest goog-object-ns
+  (testing "nbb's goog.object maps to actual goog.object"
+    (dorun (map (fn [[gobj-sym gobj-fn]]
+                  (is (ifn? gobj-fn))
+                  (is (= (str gobj-sym) (gobj/get gobj-fn "name"))))
```

Comment:
With advanced compilation this failed. It did catch a typo as I was writing the code but I might have to remove it so the tests pass in CI.